src/execution/operator/persistent/physical_insert.cpp

```diff
@@ -295,6 +296,9 @@ static idx_t PerformOnConflictAction(InsertLocalState &lstate, ExecutionContext
 			return update_chunk.size();
 		}
 		auto &local_storage = LocalStorage::Get(context.client, data_table.db);
+		D_ASSERT(gstate.initialized);
+		data_table.FinalizeLocalAppend(gstate.append_state);
```

Comment:
If we are finalising here, should we then remove the finalise from the `Sink`? Or do we need to call it twice?
```cpp
			// Flush the append so we can target the data we just appended with the update
			storage.FinalizeLocalAppend(gstate.append_state);
			gstate.initialized = false;
			(void)HandleInsertConflicts<true>(table, context, lstate, lstate.update_chunk, *this);
			(void)HandleInsertConflicts<false>(table, context, lstate, lstate.update_chunk, *this);
```

Reply:
I don't fully remember why this is here, but I can't delete it, I checked
I think it has to do with inner conflicts

If we run `OnConflictHandling` and we have inner conflicts, we save the last occurrences of these conflicts so we can run these `HandleInsertConflicts` on those tuples.

I guess in that case we don't need to flush again, so we can make it conditional on `gstate.initialized` but that's about all we can change here.

Reply:
Ah actually you're right, this flush can be removed, as it will happen inside PerformOnConflictAction already